text/0000-portals.md

```diff
@@ -0,0 +1,108 @@
+2) A way to define the destination portal of an element (eg. `lwc:portal` attribute on an element)
+
+One non-trivial example of such usage:
+
+```xhtml
+<body>
+  <div lwc:portal-definition="global-modals"></div>
+
+  <div lwc:portal-definition="popups"></div>
+  ...
+  <div id="tabs">
+    <div id="tab-1">
+      <div lwc:portal-definition="tab1-modals"></div>
+
+      <div>
+         <x-dialog lwc:portal="global-modals">Global Modal</x-foo>
+         <x-dialog lwc:portal="tab1-modals">Scoped Modal</x-foo>
+      </div>  
+    </div>  
+    <div id="tab-2">
+      <div lwc:portal-definition="tab2-modals"></div>
+
+      <div>
+         <x-dialog lwc:portal="global-modals">Global Modal</x-foo>
+         <x-dialog lwc:portal="tab2-modals">Scoped Modal</x-foo>
+      </div>  
+    </div>  
+  </div>
+</body>
+```
+since we wouldn't want for a user to specify the portal each and every time one would use a dialog, more realistically -- `<x-dialog>` would have a default behaviour based on the `@Context`, which can provide whether scoping is enabled or not, the name of the scoped portal and the name of the global portal. This dynamic choice would indicate that support for dynamically specifying the portal may be needed.
+
+## Detailed design
+
+(not an actual detailed design, just adding some possible requirements)
+
+- When a portal is defined, the name must be unique.
+- Once a portal is defined, it cannot be changed (ie. `lwc:portal` can't be removed and the name is immutable). 
+- When an element that defined a portal is removed, any element inserted into it gets added back into their natural dom position. 
+- An element can't both define a portal, and define a destination portal (ie. no `lwc:portal-define` and `lwc:portal` attributes simultaneously). 
+- Instead of being inserted at its natural position in the template, the element would be inserted in the destination portal, if one exists.
+- When an element's destination portal doesn't exist, but later is inserted in the DOM, any elements that have already rendered stay in their original location, as if the portal doesn't exist.
+- When the host element is disconnected, the portalled element should also be disconnected. 
+- The portalled component should still be `@Context` aware 
+
+## Drawbacks
+
+- new API
+- additional complexity in LWC
+
+## Security
+
+- The portalled element content should not be accessible by anyone but the originally owning host element.
+
+## Alternatives
+
+In case of dialogs, some other strategies have been considered:
+1. Event based creation of a dialog with a dialog manager defined at some parent node. 
+Given that a modal could have up to 3 slots (header, default, footer), the ergonomics of usage become very cumbersome.
+Moreover event-based modal creation is inferior to `<dialog>` like experience.
+2. Exposing top layer by the browsers. This would allow one to effectively declare a new top layer and override any 
```

Comment:
@JanMiksovsky has some questions here that we should clarify, starting with "why is scoping a thing?"

Reply:
I added some clarifying notes